test/src/main/java/org/apache/accumulo/test/functional/PermissionsIT.java

```diff
@@ -419,7 +421,14 @@ private void testMissingSystemPermission(String tableNamePrefix, AccumuloClient
         break;
       case OBTAIN_DELEGATION_TOKEN:
         if (saslEnabled()) {
```

Comment:
It doesn't appear that SASL is actually enabled in this test so this code never executes which I'm sure is why the TODO is there in the first place and why the test seems to work.

SASL is enabled by setting static system properties that are checked in `AccumuloClusterHarness` so there may be some work here to only enable it for that one check or maybe it's simplest to create a new test class (or one that extends this) that sets the property to enable it so the `TestingKdc` is initialized and this can be tested.

Reply:
Oops, good catch, guess I missed that.